src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -1030,6 +1030,49 @@ public void RemoveAnalyzerReference(string fullPath)
         }
     }
 
+    private OneOrMany<string> GetMappedAnalyzerPaths(string fullPath)
+    {
+        fullPath = Path.GetFullPath(fullPath);
+
+        if (IsSdkCodeStyleAnalyzer(fullPath))
+        {
+            // We discard the CodeStyle analyzers added by the SDK when the EnforceCodeStyleInBuild property is set.
+            // The same analyzers ship in-box as part of the Features layer and are version matched to the compiler.
+            return OneOrMany<string>.Empty;
```

Comment:
Just to call it out explicitly, doing this here will:

1. Also apply to VS Code (which might be reasonable since we'll have similar issues there I imagine)
2. Will _not_ apply to MSBuildWorkspace, which might be bad, or might not be.

The latter we can of course defer for this change (since this has huge dogfooding benefit), but wanted to call out the former.

Reply:
I really don't love that we're expanding this mechanism, given the work we're doing to allow the SDK to redirect analyzers via https://github.com/dotnet/roslyn/pull/74820 

That isn't a reason to block on this, given its not done yet, but could we open an issue to investigate if we could replace this with the new API when it's introduced?